dask/array/reductions.py

```diff
@@ -1817,3 +1817,41 @@ def nanmedian(a, axis=None, keepdims=False, out=None):
 
     result = handle_out(out, result)
     return result
+
+
+def ravel_chunks(a, out=None):
+    """
+    Flatten a dask array object chunk by chunk.
+
+    Parameters
+    ----------
+    a : dask array
+        The shape and data-type of `a` define these same attributes of the
+        returned array.
+
+    Returns
+    -------
+    out : ndarray
+        1-dimensional array of data with the same type as `a`, with shape `(a.size,)`.
+
+    See Also
+    --------
+    ravel : Return a contiguous flattened array.
+
+    Notes
+    -----
+    This function improves performances of ravel and flatten
+    operations, but does not match the initial order of the input array.
```

Comment:
Could we add a note like this to the `ravel` and `flatten` doc-strings? It would be valuable for people to know that they can improve performance if they are willing to sacrifice the initial global ordering.

Reply:
Note added to `ravel`... I cannot find where I have to add the note for `flatten`: it seems like `flatten` redirects to `ravel` anyway...?